amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/HierarchicalShardSyncer.java

```diff
@@ -96,20 +101,24 @@ public HierarchicalShardSyncer(final boolean isMultiStreamMode) {
     // CHECKSTYLE:OFF CyclomaticComplexity
     public synchronized void checkAndCreateLeaseForNewShards(@NonNull final ShardDetector shardDetector,
             final LeaseRefresher leaseRefresher, final InitialPositionInStreamExtended initialPosition,
-            final boolean cleanupLeasesOfCompletedShards, final boolean ignoreUnexpectedChildShards,
-            final MetricsScope scope) throws DependencyException, InvalidStateException,
-            ProvisionedThroughputException, KinesisClientLibIOException {
-        final List<Shard> latestShards = getShardList(shardDetector);
-        checkAndCreateLeaseForNewShards(shardDetector, leaseRefresher, initialPosition, cleanupLeasesOfCompletedShards,
-                                        ignoreUnexpectedChildShards, scope, latestShards);
+            final boolean garbageCollectLeases, final boolean cleanupLeasesOfCompletedShards,
```

Comment:
Decided to pass a flag to enable garbage collection, opted to use this instead of using something like the Decorator pattern since otherwise we need to expose the GC method